driver-core/src/main/com/mongodb/internal/connection/ServerAddressHelper.java

```diff
@@ -34,6 +34,19 @@ public static ServerAddress createServerAddress(final String host, final int por
         }
     }
 
+    /**
+     * @param address Either a host as in {@link ServerAddress#ServerAddress(String)} or both a host and a port separated with {@code ':'}.
+     */
+    public static ServerAddress parse(final String address) {
```

Comment:
This class is the second best place to put such a method: ideally, it should be in `ServerAddress`, but I anticipated that adding it there would not be accepted (after all, `InetSocketAddress` somehow exists without a similar method).

If we place the method anywhere else, e.g. in `ServerSelectionWithinLatencyWindowTest`, and in the future someone will need the same method, the method will be reinvented, because the only places where one may reasonably expect to find such a method are `ServerAddress` and `ServerAddressHelper`. Given this argument, are you still opposed to placing the method in `ServerAddressHelper`? If yes, I will move it to the test that uses the method.

Reply:
I had another look at what this method is doing, and realized that it's not necessary.  Simply calling `new ServerAddress(address)` should suffice, since that method will parse the address into host and port (though it's not even documented that it does so, it does so).

Reply:
The constructor parsing `host` is surprising given that it calls `this(host, defaultPort())`. Done.

Reply:
Agreed.